foreman/client.py

```diff
@@ -383,30 +381,46 @@ def _get_local_defs(self, strict=True):
         :param strict: Use any version that shared major version and has lower
                        minor if no total match found
         """
+        version = parse_version(self.version)
         defs_path = os.path.join(os.path.dirname(__file__), 'definitions')
         files = glob.glob('%s/*-v%s.json' % (defs_path, self.api_version))
+        files = [
+            (fn, parse_version(os.path.basename(fn).split('-')[0]))
+            for fn in files
+        ]
+
         last_major_match = None
-        for f_name in sorted(files):
-            f_ver = os.path.basename(f_name).split('-')[0]
-            if f_ver == self.version:
+        last_lower_version = None
+        for f_name, f_ver in sorted(files, key=lambda x: x[1]):
+            if f_ver == version:
                 return json.loads(open(f_name).read())
-            if f_ver.split('.')[:2] == self.version.split('.')[:2]:
+            if f_ver[:2] == version[:2]:
                 last_major_match = f_name
-            if ver_cmp(f_ver, self.version) < 0:
-                if strict:
-                    raise ForemanVersionException(
-                        "Unable to get suitable json definition for Foreman "
-                        "%s, but found a similar cached version %s/%s, run "
-                        "without strict flag to use it"
-                        % (self.version, self.api_version, last_major_match))
-                else:
-                    logging.warn("Not exact version found, found %s in cache "
-                                 "for Foreman %s", f_ver, self.version)
-                    return json.loads(open(f_name).read())
+            elif f_ver < version:
+                last_lower_version = f_name
+            else:
+                break
+
         if last_major_match:
-            logging.warn("Not exact version found, using %s from cache "
-                         "for Foreman %s", f_ver, self.version)
+            logging.warn(
```

Comment:
This is the one that should be checking for the strict, not the other one. I don't really think it's a good idea to let anyone use a lower major version definitions file, it's most likely to fail or behave unexpectedly.